### PR TITLE
add service account annotation

### PR DESCRIPTION
**Pull request by quanjielin**
to simplify deployment later. 

https://github.com/istio/istio/issues/5903

### CODE REVIEW

File: install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spec:
       annotations:
         sidecar.istio.io/inject: "false"
         scheduler.alpha.kubernetes.io/critical-pod: ""
+        cloud.google.com/service-account: "${GATEWAY_SA_NAME}"
```
[the remark this answers is not in the document]
Comment (quanjielin) [reply install/kubernetes/helm/istio/charts/gateways/templates/deployment.yaml]: already updated, refresh ?

File: samples/bookinfo/platform/kube/bookinfo.yaml [hunk 2]
```diff
@@ -21,6 +21,8 @@ metadata:
   name: details
   labels:
     app: details
+  annotations:
```
Comment (wattli) [inline samples/bookinfo/platform/kube/bookinfo.yaml]: Is it okay to have empty annotations? i.e., if a customer wants to deploy this bookinfo.yaml as it is now (without running any script to put GCP identity), would this cause any problems?

File: install/kubernetes/helm/istio/values.yaml [hunk 3]
```diff
@@ -210,7 +210,8 @@ gateways:
 
     loadBalancerIP: ""
     loadBalancerSourceRanges: {}
-    serviceAnnotations: {}
+    serviceAnnotations:
```
Comment (wattli) [inline install/kubernetes/helm/istio/values.yaml]: I am worried about the situation when customers deploy the istio-demo-auth.yaml as it is, in this case, this "serviceAnnotations" field will be empty, would this cause any issue?

Comment (quanjielin) [reply install/kubernetes/helm/istio/values.yaml]: in istio-demo-auth.yaml sds is enabled by default, so ingress->sidecar won't work leave this empty, which is expected

**Pull request merged**